Source/Turbo Navigator/TurboNavigationHierarchyController.swift

```diff
@@ -86,7 +86,10 @@ class TurboNavigationHierarchyController {
                 delegate.visit(visitable, on: .main, with: proposal.options)
             }
         case .modal:
-            if navigationController.presentedViewController != nil, !modalNavigationController.isBeingDismissed {
+            if let proposedModalNavController = controller as? UINavigationController {
+                proposedModalNavController.setModalPresentationStyle(via: proposal)
+                navigationController.present(proposedModalNavController, animated: true)
+            } else if navigationController.presentedViewController != nil, !modalNavigationController.isBeingDismissed {
```

Comment:
Unfortunately this doesn't work when navigating with a `UINavigationController` instance (e.g. `UIImagePickerController`, or `EKEventEditViewController`) while in a modal presentation.